src/t8_cmesh/t8_cmesh_cxx.cxx

```diff
@@ -150,3 +222,667 @@ t8_cmesh_uniform_bounds (t8_cmesh_t cmesh, int level, t8_scheme_cxx_t *ts, t8_gl
+  }
+
+  /* 
+   *
+   *  Cmesh is not partitioned
+   * 
+   */
+  /* If the initial cmesh is not partitioned, every process knows "everything" and we do not
+   * need any communication.*/
+  if (!cmesh->set_partition) {
+    t8_gloidx_t current_tree_element_offset;
+    const t8_gloidx_t num_trees = t8_cmesh_get_num_trees (cmesh);
+    t8_debugf ("Cmesh is not partitioned.\n");
+    /* Compute the first and last element of this process. Then loop over
+     * all trees to find the trees in which these are contained.
+     * We cast to long double and double to prevent overflow. */
+    /* cmesh is replicated, therefore the computation of local_num_children equals the global number of children*/
+    const t8_gloidx_t first_child
+      = t8_cmesh_get_first_element_of_process (cmesh->mpirank, cmesh->mpisize, local_num_children);
+    const t8_gloidx_t last_child
+      = t8_cmesh_get_first_element_of_process (cmesh->mpirank + 1, cmesh->mpisize, local_num_children) - 1;
+    t8_debugf ("[H] hybrid fc = %li, lc = %li\n", first_child, last_child);
+
+    /* Can't we optimize this linear loop by using a binary search?
+     * -> No, we cannot. Since we need in any case compute the t8_element_count_leafs_from_root
+     *    for each tree.
+     */
+    current_tree_element_offset = 0;
+    for (t8_gloidx_t igtree = 0; igtree < num_trees; ++igtree) {
+      const int ieclass = t8_cmesh_get_tree_class (cmesh, (t8_locidx_t) igtree);
+      tree_scheme = scheme->eclass_schemes[ieclass];
+      /* TODO: We can optimize by buffering the elem_in_tree value. Thus, if 
+         the computation is expensive (may be for non-morton-type schemes),
+         we do it only once. */
+      const t8_gloidx_t elem_in_tree = tree_scheme->t8_element_count_leafs_from_root (level);
+      /* Check if the first element is on the current tree */
+      if (current_tree_element_offset <= first_child && first_child < current_tree_element_offset + elem_in_tree) {
+        if (child_in_tree_begin != NULL) {
+          *child_in_tree_begin = first_child - current_tree_element_offset;
+        }
+        *first_local_tree = igtree;
+        /* If our first element is not the very first element in the tree, we share
+         * this tree with the previous process. */
+        if (first_tree_shared != NULL) {
+          *first_tree_shared = current_tree_element_offset < first_child ? 1 : 0;
+        }
+      }
+      if (last_child < first_child) {
+        /* This process is empty. We needed to identify the 'first_local_tree' since it is the 
+         * first local tree of the next process, which we store in this case.
+         */
+        t8_debugf ("[D] lc < fc\n");
+        t8_cmesh_uniform_set_return_parameters_to_empty (first_local_tree, child_in_tree_begin, last_local_tree,
+                                                         child_in_tree_end, first_tree_shared);
+        return;
+      }
```

Comment:
The function call to `t8_cmesh_uniform_set_return_parameters_to_empty ` does not do what the comment above promises.
I think that this is the root of the offset problems that we see.

Reply:
```suggestion
      if (last_child < first_child) {
        /* This process is empty. We needed to identify the 'first_local_tree' since it is the 
         * first local tree of the next process, which we store in this case.
         */
        t8_debugf ("[D] lc < fc\n");
        t8_cmesh_uniform_set_return_parameters_to_empty (first_local_tree, child_in_tree_begin, last_local_tree,
                                                         child_in_tree_end, first_tree_shared);
        *first_local_tree = 0;
        return;
      }
```
Using this actually enforces `num_local_elements = 0` further down. This fixes the case, where the original mesh was replicated, but in the same the test, the cmesh gets repartitioned again, so now there is an error further down in the partitioned part of this function